addons/hr_timesheet/models/project.py

```diff
@@ -80,18 +80,26 @@ def _search_is_internal_project(self, operator, value):
             operator_new = 'not inselect'
         return [('id', operator_new, (query, ()))]
 
-    @api.depends('allow_timesheets', 'task_ids.planned_hours', 'task_ids.remaining_hours')
+    @api.depends('allow_timesheets', 'task_ids.planned_hours', 'timesheet_ids')
     def _compute_remaining_hours(self):
-        group_read = self.env['project.task'].read_group(
-            domain=[('planned_hours', '!=', False), ('project_id', 'in', self.filtered('allow_timesheets').ids),
+        group_read_effective_hours = self.env['account.analytic.line'].read_group(
+            domain=[('project_id', 'in', self.filtered('allow_timesheets').ids),
+                    ('task_id', '!=', False), ('task_id.planned_hours', '!=', 0.0),
+                    '|', ('task_id.stage_id.fold', '=', False), ('task_id.stage_id', '=', False)],
+            fields=['effective_hours:sum(unit_amount)'], groupby='project_id')
```

Comment:
No need to name the parameter if you keep the same order than the one defined in the method. 🙂 
Also, I prefer naming the variable containing the result of read_group like `{model_name}_read_group`. Here it is timesheet then we could name it `timesheet_read_group`.
```suggestion
        timesheet_read_group = self.env['account.analytic.line'].read_group(
            [
                ('project_id', 'in', self.filtered('allow_timesheets').ids),
                ('task_id', '!=', False), 
                ('task_id.planned_hours', '!=', 0.0),
                '|', ('task_id.stage_id.fold', '=', False), ('task_id.stage_id', '=', False)],
            ['effective_hours:sum(unit_amount)'], 
            'project_id'
        )
```
Moreover, I don't understand why you just take the timesheets in opened tasks. 🤔

Reply:
I guess you add this field because we use the `task_ids` in the dependencies instead of using `tasks` field. I think we want to have the total value for **all tasks** linked to the project and we could assume only the planned_hours in the opened tasks should be changed, that's why it is not `tasks.planned_hours` in the dependencies of that method.

To confirm with @aju-odoo . 🙂

Reply:
Hello @nboulif , I discussed with @aju-odoo about that and she wants to take into account the open tasks, so I guess it is good as you did. 🤔